File: LeetCode/cpp/703.cpp
class KthLargest {
    priority_queue<int, vector<int>, greater<int>> p;
    int k;
public:
    KthLargest(int k, vector<int> nums) {
        this->k = k;
        for (auto m: nums) {
            if (p.size() < k || p.top() < m)
                p.push(m);
            if (p.size() > k)
                p.pop();
        }
    }

    int add(int val) {
        if (p.size() < k || p.top() < val)
            p.push(val);
        if (p.size() > k)
            p.pop();
        return p.top();
    }
};

File: LeetCode/python3/703.py
class KthLargest:

    def __init__(self, k: 'int', nums: 'List[int]'):
        self.k = k
        self.h = nums
        heapq.heapify(self.h)
        while len(self.h) > k:
            heapq.heappop(self.h)

    def add(self, val: 'int') -> 'int':
        heapq.heappush(self.h, val)
        if len(self.h) > self.k:
            heapq.heappop(self.h)
        return self.h[0]